BootloaderCorePkg/Library/MpInitLib/Ia32/MpInitLib.c

```diff
@@ -16,6 +16,59 @@ volatile MP_DATA_EXCHANGE_STRUCT   mMpDataStruct;
 UINT8                             *mBackupBuffer;
 UINT32                             mMpInitPhase = EnumMpInitNull;
 
+/**
+  Sort the CPU entry according to their thread distances
+
+  It is required to list CPU thread with further distance first so as to
+  fully utilize unshared CPU resources.
+
+  @param[in]  SysCpuInfo     Pointer to the ALL_CPU_INFO structure.
+
+**/
+STATIC
+VOID
+SortSysCpu (
```

Comment:
What about adding basic sort lib? We are using sort in pci enumeration and shell commands. I think insertion sort can be used as common.

Reply:
can we do this in a separate commit ?

Reply:
Sure. I think a separate PR would be better to cover all cases.